osmtm/static/js/project.js

```diff
@@ -766,7 +766,6 @@ osmtm.project = (function() {
 
   function checkForUpdates() {
     window.clearTimeout(checkTimeout);
```

Comment:
In theory, this should be called only in two places - document.ready/onload; and from a success or failure in checkForUpdates; so you probably don't need to do the ```clearTimeout``` if you can guarantee that.

If you can't / want to guard against it being invoked twice in close succession; you probably want to track the state of the underlying ajax request and abort it (https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/readyState) when you clear.

Not a huge deal unless you do something like ```for (i=0; i< 5000; i++) { checkForUpdates(); }``` I suppose. 

It feels a bit mixed at the moment, some of the code is handling global (window) state; other bits are just "I execute when I'm  told to" - better splitting them up might make things a bit easier to reason about